packages/pigeon/example/app/ios/Runner/Messages.g.swift

```diff
@@ -26,19 +26,50 @@ private func wrapError(_ error: Any) -> [Any?] {
       flutterError.details,
     ]
   }
+  if let pigeonError = error as? PigeonError {
+    return [
+      pigeonError.code,
+      pigeonError.message,
+      pigeonError.details,
+    ]
+  }
   return [
     "\(error)",
     "\(type(of: error))",
     "Stacktrace: \(Thread.callStackSymbols)",
   ]
 }
 
-private func createConnectionError(withChannelName channelName: String) -> FlutterError {
-  return FlutterError(
+private func createConnectionError(withChannelName channelName: String) -> PigeonError {
+  return PigeonError(
     code: "channel-error", message: "Unable to establish connection on channel: '\(channelName)'.",
     details: "")
 }
 
+/// Error class for passing custom error details to Flutter.
+final class PigeonError: Error {
+  let code: String
+  let message: String?
+  let details: Any?
+
+  init(code: String, message: String?, details: Any?) {
+    self.code = code
+    self.message = message
+    self.details = details
+  }
+
+  init(flutterError: FlutterError) {
```

Comment:
why do we need this?

Reply:
Please remove if it's not used. I don't see why people still have to use `FlutterError`. Otherwise LGTM.